pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -580,6 +558,50 @@ func (l *LibvirtDomainManager) asyncMigrate(vmi *v1.VirtualMachineInstance, opti
 	}(l, vmi)
 }
 
+// startMigrationProxyServer starts a local migration proxy.
+// Right now Libvirt won't let us perform a migration using a unix socket, so
+// we have to create a local host tcp server (on port 22222) that forwards the traffic
+// to libvirt in order to trick libvirt into doing what we want.
+// This also creates a tcp server for each additional direct migration connections
+// that will be proxied to the destination pod
+func startMigrationProxyServer(virtShareDir string, vmi *v1.VirtualMachineInstance, loopbackAddress string) (stopServer func(), err error) {
+	var proxyStopListenPool []func()
+	stopServer = func() {
+		for _, f := range proxyStopListenPool {
+			f()
+		}
+	}
+	defer func() {
+		// On error, stop the servers before returning.
+		if err != nil {
+			stopServer()
+			stopServer = nil
+		}
+	}()
+
+	isBlockMigration := vmi.Status.MigrationMethod == v1.BlockMigration
+	migrationPortsRange := migrationproxy.GetMigrationPortsList(isBlockMigration)
+
+	// Create a tcp server for each direct connection proxy
+	for _, port := range migrationPortsRange {
+		key := migrationproxy.ConstructProxyKey(string(vmi.UID), port)
+		migrationProxy := migrationproxy.NewTargetProxy(loopbackAddress, port, nil, nil, migrationproxy.SourceUnixFile(virtShareDir, key))
+		if e := migrationProxy.StartListening(); e != nil {
+			return stopServer, e
```

Comment:
Previously the `StopListening` was invoked even in case of an error. Are you sure the is nothing to clean/stop in case of an error?
If so, please updated the commit message that the commit is not just and extract of a method.

Reply:
I mean that in case of a error, you return before appending the `StopListening` method to the `proxyStopListenPool`.
It means `StopListening` won't be called on a error migration proxy.
Before this commit, it was invoked also in case of an error.

Reply:
Yes, if an error occurs, the expectation is that there is nothing to cleanup.
Looking into the existing function, the error may occur only to create the socket, so I could not see any issue here.

I will update the commit message.

Reply:
Done